src/source-manager.ts

```diff
@@ -137,7 +137,7 @@ export class SourceManager {
     try {
       // For now, use GitHub API (in future this could support local files, git clones, etc.)
       const repoUrl = source.repo;
-      const match = repoUrl.match(/github\.com\/([^\/]+)\/([^\/]+)(?:\.git)?/);
+      const match = repoUrl.match(/github\.com\/([^\/]+)\/([^\/]+?)(?:\.git)?$/);
```

Comment:
After brushing up on regex, I see here we're extracting the github account and repository name - as a clarification for my own regex knowledge, why do we need the question mark for the second extraction pattern?

Reply:
Oh nice, yeah the option to point to local files would be nice for testing. Thanks for the detailed explanation! 👍